modules/packages/BLAS.chpl

```diff
@@ -695,8 +695,662 @@ module BLAS {
+       if p[0] == -2 then  H = |     |
+                               |0   1|
+
+
+
+*/
+proc rotmg(ref d1: ?eltType, ref d2: eltType, ref b1: eltType, b2: eltType, P: []eltType) {
+  select eltType {
+    when real(32) do{
+      cblas_srotmg(d1,d2,b1,b2,P);
+    }
+    when real(64) do{
+      cblas_drotmg(d1,d2,b1,b2,P);
+    }
+    otherwise {
+        compilerError("Unknown type in rotmg");
+    }
+  }
+}
+
+/*
+    Wrapper for the `ROT routines <http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1_ga0ce1ab4726eb7ad925cbd89f100d5ce0.html#ga0ce1ab4726eb7ad925cbd89f100d5ce0>`_
+
+    Replaces the value elements of two  vectors X and Y using the equation::
+
+      X[i] = c*X[i] + s*X[i]
+      Y[i] = c*Y[i] - s*X[i]
+
+    Input:
+    X,Y: Vectors with N elements.
+    c,s: Scalar parameters.
+    incY: defines the increment for the vector Y.
+    incX: defines the increment for the vector X.
+
+    Output:
+    X: Vector with updated elements calculated by ``X[i] = c*X[i] + s*X[i]``
+    Y: Vector with updated elements calculated by ``Y[i] = c*Y[i] - s*X[i]``
+
+
+*/
+proc rot(X: [?D] ?eltType, Y: [D] eltType, c: eltType, s: eltType,  incY: c_int = 1, incX: c_int = 1)
+where D.rank == 1 {
+
+  const N = D.size: c_int;
+
+  select eltType {
+    when real(32) do{
+      cblas_srot(N, X, incX, Y, incY, c, s);
+    }
+    when real(64) do{
+      cblas_drot(N, X, incX, Y, incY, c, s);
+    }
+    otherwise {
+        compilerError("Unknown type in rot");
+    }
+  }
+}
+/*
+    Wrapper for the `ROTM routines  <http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1_ga5633344a5729b4f0167aa441dcf95a9c.html#ga5633344a5729b4f0167aa441dcf95a9c>`_
+
+    Executes the modified `Givens rotations <https://en.wikipedia.org/wiki/Givens_rotation>`_ with element wise  substitution::
+
+       |X[i]|     |X[i]|
+       |    | = H.|    |
+       |Y[i]|     |Y[i]|
+
+    Input:
+    X,Y: Vectors with N elements.
+    incY: defines the increment for the vector Y.
+    incX: defines the increment for the vector X.
+    P  : P is an 5 element array parameter.
+
+    p[0] : It is a flag parameter with possible values::
+
+                               |p[1] p[3]|
+       if p[0] == -1 then  H = |         |
+                               |p[2] p[4]|
+
+                               |1.0  p[3]|
+       if p[0] ==  0 then  H = |         |
+                               |p[2]  1.0|
+
+                              |p[1]  1.0|
+       if p[0] == 1 then  H = |         |
+                              |-1.0 p[4]|
+
+                               |1   0|
+       if p[0] == -2 then  H = |     |
+                               |0   1|
+
+     Output:
+     X: Transformed X vector with elements updated by : ``X[i] = p[1]*X[i] +p[3]*Y[i]``
+     Y: Transformed Y vector with elements updated by : ``Y[i] = p[2]*X[i] +p[4]*Y[i]``
+
+
+*/
+proc rotm(X: [?D]?eltType,  Y: [D]eltType,  P: []eltType, incY: c_int = 1, incX: c_int = 1)
```

Comment:
I think this is already accomplished nicely through the domain type query behavior. Passing arrays of mismatched domains results in the following error:

```
$CHPL_HOME/modules/packages/BLAS.chpl:823: error: halt reached - Domain mismatch passing array argument:
  Formal domain is: {0..1}
  Actual domain is: {0..2}
```